pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -74,6 +74,7 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 
 	s3svc, err := cephObject.NewS3Agent(p.accessKeyID, p.secretAccessKey, p.getObjectStoreEndpoint(), true)
 	if err != nil {
+		p.deleteOBCResourceLogError("")
```

Comment:
Can't we just `defer` instead of these multiple calls?

Reply:
Yes, but we only want to do this on error... I suppose we could do what the lib-bucket-provisioner code does and have an `if err != nil` block in the deferred function...

Reply:
Also given that some of the cleanup calls use `p.bucketName` I'd rather duplicate the lines and be more clear rather than have a complicated cleanup call. But let me know if you still think we should defer cleanup.